pkg/tests/tasks/traffic/ingress/gatewayapi_test.go

```diff
@@ -0,0 +1,164 @@
+package ingress
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+func TestGatewayApiControllerMode(t *testing.T) {
+	NewTest(t).Id("T41").Groups(Full, InterOp).Run(func(t TestHelper) {
+		if env.GetSMCPVersion().LessThan(version.SMCP_2_3) {
+			t.Skip("TestGatewayApiControllerMode was added in v2.3")
+		}
+		ns := "foo"
```

Comment:
You should use a namespace that is *not* part of the mesh, since the Gateway API must work correctly in non-member namespaces.

Reply:
I have tested with non-members and the gateway is waiting for the controller.

Reply:
Once, we add the namespace to Member. It has a successful connection.